ggshield/core/types.py

```diff
@@ -18,8 +18,8 @@ def filter_fields(cls, data: Dict, **kwargs: Any) -> Dict:
         field_names = {field_.name for field_ in fields(cls)}
         filtered_fields = {}
         for key, item in data.items():
-            if key in field_names:
-                filtered_fields[key] = item
+            if key in field_names or key.replace("-", "_") in field_names:
+                filtered_fields[key.replace("-", "_")] = item
```

Comment:
I would change this to:

```suggestion
            filtered_key = key.replace("-", "_")
            if filtered_key in field_names:
                filtered_fields[filtered_key] = item
```

It's a bit simpler and avoid performing the `replace()` call twice.

Reply:
Done!